### DISCUSSION TITLE
UAParser.js changes the license model in version 2

### DISCUSSION THREAD

**Discussion by reneoelke** [grafana/faro-web-sdk#624] [General]
The package UAParser.js (ua-parser-js) will change its [licence model from version 2](https://github.com/faisalman/ua-parser-js/discussions/674). From MIT to AGPL (for open source use) or PRO (for commercial use).

Version "^1.0.32" of ua-parser-js is used in the faro-web-sdk package. Presumably an upgrade to version 2 will not be an option, as AGPL is not compatible with the Apache licence. There will probably be updates for the older MIT version 1.x for a while. However, no exact schedule has been defined.

Does Grafana (faro-web-sdk) already have a plan for future versions of ua-parser-js?

**Comment by codecapitano** [comment]
Hi @reneoelke good catch wasn't aware about this change, thanks a lot!

We'll double check future plans/options and I'll update this thread.

Cheers,
Marco

**Comment by codecapitano** [comment]
Hi @reneoelke we do not integrate software under AGPL into Faro.
For now we'll keep to pin the version to stay < v2 and plan what to do in the future. 

Thanks again for bringing this up.

Cheers,
Marco

**Comment by reneoelke** [comment]
Thanks for your information @codecapitano
